test/apps/data-filter/src/scatterplot-layer.js

```diff
@@ -0,0 +1,38 @@
+import {ScatterplotLayer} from '@deck.gl/layers';
+import dataFilter from './data-filter';
+
+const defaultProps = {
+  getFilterValue: 1
+};
+
+export default class ScatterplotLayerWithFilter extends ScatterplotLayer {
+
+  getShaders() {
+    const shaderSettings = super.getShaders();
+
+    shaderSettings.modules.push(dataFilter);
+
+    shaderSettings.inject = {
+      'vs:#decl': `
```

Comment:
It would look so much crisper if we didn't need the newlines here. Maybe the could automatically be added, or inject keys could have a flag that adds them, or prevents them from being added?